### PR TITLE
Fix accuracy of Ambient probe Lighting and Emissive Property

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
Following the question from a VFX supervisor, I have found some incorrect code in our "color pipeline".
This PR:
- Fix Ambient occlusion probe accuracy, was incorrect (9% error)
- Fix color space of EmissiveLDR which should be convert to linear space
- Clean lot of useless code in Emissive and removed "normalization" option which where hidden anyway.

---
### Testing status

Tested:
- Lit  interface of EmissiveColor with both HDR and use intensity case, checked the result in the .mat. Do the same for unlit.
All shader share the same code so no need to test lit tessellation, layered lit etc...
- check the shader graph emission node, check that it correctly match unlit and lit value (checked with the color picker the resulting value). 
- Inspected all mat files that where upgrade to ensure upgrade is correct
- No futher test needed, Yamato handle it.

Various test will have slight change in the sky lighting due to the fix. I have updated the screenshots
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/HDRP%252Ffix-color-pipeline/.yamato%252F_abv.yml%2523all_project_ci_fast-trunk/2942993/job/pipeline

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/AssetProcessors/MaterialPostProcessor.cs [hunk 1]
```diff
@@ -561,6 +562,32 @@ static void ExposedDecalInputsFromShaderGraph(Material material, HDShaderUtils.S
...
+                const string kEmissiveColorLDR = "_EmissiveColorLDR";
+                const string kEmissiveColor = "_EmissiveColor";
+                const string kEmissiveIntensity = "_EmissiveIntensity";
+
+                if (material.HasProperty(kEmissiveColorLDR) && material.HasProperty(kEmissiveIntensity) && material.HasProperty(kEmissiveColor))
+                {
+                    // Important:  The color picker for kEmissiveColorLDR is LDR and in sRGB color space but Unity don't perform any color space conversion in the color
+                    // picker BUT only when sending the color data to the shader... So as we are doing our own calculation here in C#, we must do the conversion ourselves.
+                    Color emissiveColorLDR = material.GetColor(kEmissiveColorLDR);
+                    Color emissiveColorLDRLinear = new Color(Mathf.LinearToGammaSpace(emissiveColorLDR.r), Mathf.LinearToGammaSpace(emissiveColorLDR.g), Mathf.LinearToGammaSpace(emissiveColorLDR.b));
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Editor/AssetProcessors/MaterialPostProcessor.cs:584]: This is named emissiveColorLDRLinear, but we are converting **to** gamma **from** linear

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/AssetProcessors/MaterialPostProcessor.cs:584]: good catch :)

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/AssetProcessors/MaterialPostProcessor.cs]: fixed

File: com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute [hunk 2]
```diff
@@ -12,12 +12,12 @@ TEXTURECUBE(_AmbientProbeInputCubemap);
 
 // Constants from SetSHEMapConstants function in the Stupid Spherical Harmonics Tricks paper:
 // http://www.ppsloan.org/publications/StupidSH36.pdf
-#define sqrtPI (sqrt(PI))
-#define fC0 (0.282095 * (1.0 / (2.0 * sqrtPI)))
-#define fC1 (0.488603 * (sqrt ( 3.0) / ( 3.0 * sqrtPI)))
-#define fC2 (1.092548 * (sqrt (15.0) / ( 8.0 * sqrtPI)))
-#define fC3 (0.315392 * (sqrt ( 5.0) / (16.0 * sqrtPI)))
-#define fC4 (0.546274 * 0.5 * fC2)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute:20]: @JulienIgnace-Unity : this one was incorrect
#define fC4 (0.546274 * 0.5 * fC2)

it must be #define fC4 (0.546274 * 0.5 * (sqrt (15.0) / ( 8.0 * sqrtPI))) without the 1.092548

File: com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute [hunk 3]
```diff
@@ -12,12 +12,12 @@ TEXTURECUBE(_AmbientProbeInputCubemap);
...
+//  [SH basis coeff] * [clamped cosine convolution factor]
+#define fC0 (rsqrt(PI * 4.0) * rsqrt(PI * 4.0)) // Equivalent to (0.488603 * (sqrt ( 3.0) / ( 3.0 * sqrtPI)))
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute:16]: This is not equivalent to what the comment says. 

(rsqrt(PI * 4.0) * rsqrt(PI * 4.0)) == 0.0795774
(0.488603 * (sqrt ( 3.0) / ( 3.0 * sqrtPI))) == 0.159155102


So there is a 0.5 factor missing from either the factor or the comment  (I assume in the comment)

File: com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute [hunk 4]
```diff
@@ -12,12 +12,12 @@ TEXTURECUBE(_AmbientProbeInputCubemap);
...
+#define fC1 (rsqrt(PI * 4.0 / 3.0) * rsqrt(PI * 3.0)) // Equivalent to (1.092548 * (sqrt (15.0) / ( 8.0 * sqrtPI)))
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute:17]: Same here, the 

(rsqrt(PI * 4.0 / 3.0) * rsqrt(PI * 3.0))  == 0.1591549 ... 
(1.092548 * (sqrt (15.0) / ( 8.0 * sqrtPI))) = 0.2984154...

File: com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute (hunk 3 above)
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute:16]: another good catch :) . Fixed the comment

File: com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute (hunk 4 above)
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute:17]: yup, thakns

File: com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute [hunk 5]
```diff
@@ -12,12 +12,12 @@ TEXTURECUBE(_AmbientProbeInputCubemap);
...
+#define fC2 (rsqrt(PI * 4.0 / 15.0) * rsqrt(PI * 64.0 / 15.0)) // Equivalent to (1.092548 * (sqrt (15.0) / ( 8.0 * sqrtPI)))
+#define fC3 (rsqrt(PI * 16.0 / 5.0) * rsqrt(PI * 256.0 / 5.0)) // Equivalent to (0.315392 * (sqrt ( 5.0) / (16.0 * sqrtPI)))
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute:19]: here is off too 

In general I added a desmos comments vs factors  https://www.desmos.com/calculator/xrn75bj2gz

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute:19]: for fc3 you missed a PI in desmos, now all match

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute:19]: Ah right! My bad :D

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Sky/AmbientProbeConvolution.compute:19]: Was reported by a user in March but have forget and just rethinked about it :)

File: com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md [hunk 6]
```diff
@@ -42,6 +42,8 @@ From Unity 2020.2, the light layer properties have move from the HDRP settings t
 
 From Unity 2020.2, in physically based sky, the sun disk intensity is proportional to its size. Before this the sun disk was incorrectly not drive by the sun disk size.
 
+From Unity 2020.2, EmissiveColorLDR property when using the UseEmissiveIntensity property is not correctly handle as sRGB color space in Decal, Lit, LayeredLit and Unlit shader. Previously the EmissiveColor was handled as linear RGB color space. A migration script handle the conversion automatically to keep visual matching.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md:45]: From Unity 2020.2, if you previously used the **UseEmissiveIntensity** property in either a Decal, Lit, LayeredLit, or Unlit Material, be aware that the **EmissiveColorLDR** property is in sRGB color space. Previously HDRP handled **EmissiveColor** as being in linear RGB color space so there may be a mismatch in visuals between versions. To fix this mismatch, HDRP includes a migration script that handles the conversion automatically.

**Pull request merged**